src/functions/env.ts

```diff
@@ -191,49 +219,70 @@ function parseStrict(data: string): Record<string, string> {
  *
  * @return {Record<string, string>} Environment variables for the project.
  */
-export function load(options: {
+export function loadUserEnvs({
+  functionsSource,
+  projectId,
+  projectAlias,
+}: {
   functionsSource: string;
   projectId: string;
   projectAlias?: string;
 }): Record<string, string> {
-  const targetFiles = [".env"];
-  targetFiles.push(`.env.${options.projectId}`);
-  if (options.projectAlias && options.projectAlias.length) {
-    targetFiles.push(`.env.${options.projectAlias}`);
-  }
+  const envFiles = findEnvfiles(functionsSource, projectId, projectAlias);
 
-  const targetPaths = targetFiles
-    .map((f) => path.join(options.functionsSource, f))
-    .filter(fs.existsSync);
-
-  // Check if both .env.<project> and .env.<alias> exists.
-  if (targetPaths.some((p) => path.basename(p) === `.env.${options.projectId}`)) {
-    if (options.projectAlias && options.projectAlias.length) {
-      for (const p of targetPaths) {
-        if (path.basename(p) === `.env.${options.projectAlias}`) {
-          throw new FirebaseError(
-            `Can't have both .env.${options.projectId} and .env.${options.projectAlias}> files.`
-          );
-        }
-      }
+  // Disallow setting both .env.<projectId> and .env.<projectAlias>
+  if (projectAlias) {
+    if (envFiles.includes(`.env.${projectId}`) && envFiles.includes(`.env.${projectAlias}`)) {
+      throw new FirebaseError(`Can't have both .env.${projectId} and .env.${projectAlias}> files.`);
```

Comment:
I wonder if we should make it clear that they can't have those files because they can't load files by both project ID and project alias. I'd be confused if I was told `Can't have both .env.inlined-junkdrawer and .env.dev> files`

Also, is the ">" a typo?

Reply:
Yes on typo. Agreed on the confusion - added more detailed message.